### PR TITLE
Eahw 2698/restapi helm chart

### PR DESCRIPTION
**Pull request by taomoh**
Add Ingress, Service and Network Policy setup to base Helm chart.
Configuration for the following aspects was made optional to accommodate for different requirements:
- Ingress, Service and Network Policy
- Kafka and Database interaction

### CODE REVIEW

File: charts/callisto-base-service/Chart.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ description: A Helm chart for a base Callisto service
 # Library charts provide useful utilities or functions for the chart developer. They're included as
 # a dependency of application charts to inject those utilities and functions into the rendering
 # pipeline. Library charts do not define any templates and therefore cannot be deployed.
-type: application
+#type: library
```
Comment (taomoh) [inline charts/callisto-base-service/Chart.yaml:14]: Should this be reverted to application?

File: charts/callisto-base-service/templates/_helpers.tpl [hunk 2]
```diff
@@ -58,10 +58,17 @@ Create the name of the service account to use
 {{- end }}
 
 {{/*
-Image name and tag for the Callisto component being deployed
+Image name and tag for the main Callisto container being deployed
 */}}
-{{- define "callisto-base-service.deploymentImageName" -}}
-{{- required ".Values.image.repository is required" .Values.image.repository }}:{{ required ".Values.image.tag is required" .Values.image.tag }}
+{{- define "callisto-base-service.containerImageName" -}}
```
Comment (taomoh) [inline charts/callisto-base-service/templates/_helpers.tpl:63]: Should this snippet be called `mainContainerImageName?

File: charts/callisto-base-service/templates/_helpers.tpl [hunk 3]
```diff
@@ -58,10 +58,17 @@ Create the name of the service account to use
...
+{{ .Values.containerRegistryUrl }}{{- required ".Values.mainContainerImage.repositoryName is required" .Values.mainContainerImage.repositoryName }}:{{ required ".Values.mainContainerImage.tag is required" .Values.mainContainerImage.tag }}
+{{- end }}
+
+{{/*
+Image name and tag for the database migration init container
+*/}}
+{{- define "callisto-base-service.databaseMigrationImageName" -}}
+{{ .Values.containerRegistryUrl }}{{- required ".Values.databaseImage.repositoryName is required" .Values.databaseImage.repositoryName }}:{{ required ".Values.databaseImage.tag is required" .Values.databaseImage.tag }}
```
Comment (taomoh) [inline charts/callisto-base-service/templates/_helpers.tpl:71]: should the Helm value be called databaseMigrationImage

File: charts/callisto-base-service/templates/ingress.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    ingress.kubernetes.io/force-ssl-redirect: "true"
+    kubernetes.io/ingress.class: "nginx-internal"
+    kubernetes.io/backend-protocol: "HTTPS"
+    ingress.kubernetes.io/enable-cors: "true"
+    ingress.kubernetes.io/cors-allow-origin: {{ .Values.ingress.cors_origin }}
+spec:
+  tls:
+    - hosts:
+        - {{ .Values.ingress.host }}
+      secretName: {{ .Values.ingress.tls_secret_name }}
```
Comment (taomoh) [inline charts/callisto-base-service/templates/ingress.yaml:21]: camel case: `tlsSecretName`?

File: charts/callisto-base-service/templates/ingress.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  labels:
+    app: {{ include "callisto-base-service.fullname" . }}
+    cert-manager.io/solver: route53
+  annotations:
+    cert-manager.io/enabled: "true"
+    ingress.kubernetes.io/force-ssl-redirect: "true"
+    kubernetes.io/ingress.class: "nginx-internal"
+    kubernetes.io/backend-protocol: "HTTPS"
+    ingress.kubernetes.io/enable-cors: "true"
+    ingress.kubernetes.io/cors-allow-origin: {{ .Values.ingress.cors_origin }}
```
Comment (taomoh) [inline charts/callisto-base-service/templates/ingress.yaml:16]: camel case: `corsOrigin`

File: values.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
+#TODO: DO NOT MERGE TO MAIN
```
Comment (taomoh) [inline values.yaml:1]: This file should be deleted before merge to `main`

File: charts/callisto-base-service/templates/ingress.yaml [hunk 7]
```diff
@@ -0,0 +1,33 @@
+{{- if .Values.ingress }}
```
Comment (taomoh) [inline charts/callisto-base-service/templates/ingress.yaml:1]: Add a guard to only deploy ingress for branch deployment, as ingress for higher environments is managed in `callisto-ingress-nginx`

File: README.md
Comment (taomoh) [inline README.md:1]: Add values.yaml to README.md as embedded, attached, or separate readme file

**Review by joannafawl — APPROVED** [review]

**Pull request merged**